lib/flatfile/csvInformationObjectExport.class.php

```diff
@@ -0,0 +1,390 @@
+    $this->setNoteColumns();
+    $this->setRadGeneralMaterialDesignationColumn();
+    $this->setAlternativeIdentifierColumns();
+    $this->setPhysicalObjectColumns();
+    $this->setAccessionNumberColumn();
+    $this->setCreationColumns();
+    $this->setEventColumns();
+
+    // Set level of description
+    $this->setColumn(
+      'levelOfDescription',
+      $this->levelOfDescriptionTerms[$this->resource->levelOfDescriptionId]
+    );
+
+    // Set level of detail
+    $this->setColumn(
+      'levelOfDetail',
+      $this->levelOfDetailTerms[$this->resource->descriptionDetailId]
+    );
+
+    // Set description status
+    $this->setColumn(
+      'descriptionStatus',
+      $this->descriptionStatusTerms[$this->resource->descriptionStatusId]
+    );
+
+    // Set publication status
+    $this->setColumn('publicationStatus', $this->resource->getPublicationStatus());
+  }
+
+  /*
+   * Set access point-related columns
+   *
+   * @return void
+   */
+  protected function setAccessPointColumns()
+  {
+    // Set name access point columns
+    $nameAccessPointData = $this->getNameAccessPointData();
+    $this->setColumn('nameAccessPoints', $nameAccessPointData['names']);
+    $this->setColumn('nameAccessPointHistories', $nameAccessPointData['histories']);
+
+    // Set place access point columns
+    $placeAccessPointData = $this->getPlaceAccessPoints();
+    $this->setColumn('placeAccessPoints', $placeAccessPointData['names']);
+
+    // Set subject access point columns
+    $subjectAccessPointData = $this->getSubjectAccessPoints();
+    $this->setColumn('subjectAccessPoints', $subjectAccessPointData['names']);
+  }
+
+  /*
+   * Set note-related columns
+   *
+   * @return void
+   */
+  protected function setNoteColumns()
+  {
+    $this->setNoteColumnsUsingMap($this->commonNoteMap, $this->commonNoteTypeIds);
+    $this->setNoteColumnsUsingMap($this->radNoteMap, $this->radNoteTypeIds);
+    $this->setNoteColumnsUsingMap($this->titleNoteMap, $this->titleNoteTypeIds);
+  }
+
+  /*
+   * Set RAD standard's general material designation column
+   *
+   * @return void
+   */
+  protected function setRadGeneralMaterialDesignationColumn()
+  {
+    $relations = $this->resource->getTermRelations(QubitTaxonomy::MATERIAL_TYPE_ID);
+
+    if (count($relations))
+    {
+      $materials = array();
+
+      foreach($relations as $relation)
+      {
+        array_push($materials, $relation->term->name);
+      }
+
+      $this->setColumn('radGeneralMaterialDesignation', $materials);
+    }
+  }
+
+  /*
+   * Set alternative identifier-related columns
+   *
+   * @return void
+   */
+  protected function setAlternativeIdentifierColumns()
+  {
+    $properties = $this->resource->getProperties(null, 'alternativeIdentifiers');
+
+    $alternativeIdentifiers      = array();
+    $alternativeIdentifierLabels = array();
+
+    foreach($properties as $property)
+    {
+      array_push($alternativeIdentifierLabels, $property->name);
```

Comment:
http://php.net/manual/en/function.array-push.php seems to indicate in the note if you're just pushing a single value to an array, $ary[] = 123 is slightly better. But I don't think it's a big deal.